spring-cloud-dataflow-docs/src/main/asciidoc/task-developer-guide.adoc

```diff
@@ -0,0 +1,355 @@
+It also shows how to map these applications into Spring Cloud Data Flow and deploy them.
+
+[[task-dev-guide-prebuilt-apps]]
+== Prebuilt Applications
+
+The link:http://cloud.spring.io/spring-cloud-task-app-starters/[Spring Cloud Task App Starters] project provides many applications that you can start using right away.
+For example, there is a timestamp application that prints the timestamp to the console.
+All the applications are based on link:https://projects.spring.io/spring-boot/[Spring Boot] and link:https://cloud.spring.io/spring-cloud-task/[Spring Cloud Task].
+
+Applications are published as Maven artifacts as well as Docker images.
+For GA releases, the Maven artifacts are published to Maven central and the link:http://repo.spring.io/release[Spring Release Repository].
+Milestone and snapshot releases are published to the link:http://repo.spring.io/milestone[Spring Milestone] and link:http://repo.spring.io/snapshot[Spring Snapshot] repositories, respectively.
+Docker images are pushed to link:https://hub.docker.com/u/springcloudtask/[Docker Hub]
+
+The root location of the Spring Repository that hosts the GA artifacts of prebuilt applications is http://repo.spring.io/release/org/springframework/cloud/task/app/
+
+[[task-dev-guide-running-prebuilt-apps]]
+== Running Prebuilt Applications
+
+In this example, we use RabbitMQ as the messaging middleware.
+Follow the directions on link:https://www.rabbitmq.com/download.html[rabbitmq.com] for your platform.
+Then install the link:https://www.rabbitmq.com/management.html[management plugin].
+
+You can run the timestamp application by using `java -jar`.
+
+To get started, download the sample application, as follows:
+
+[source,bash]
+wget https://repo.spring.io/libs-release/org/springframework/cloud/task/app/timestamp-task/1.3.0.RELEASE/timestamp-task-1.3.0.RELEASE.jar
+
+That file contains a Spring Boot applications that includes the link:http://docs.spring.io/spring-boot/docs/current/reference/html/production-ready.html[Spring Boot Actuator] and the link:http://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-security.html[Spring Security Starter].
+You can specify link:https://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html[common Spring Boot properties] to configure each application.
+
+Now you can run the timestamp application, as follows:
+
+[source,bash]
+java -jar timestamp-task-1.3.0.RELEASE.jar --logging.level.org.springframework.cloud.task=DEBUG
+
+NOTE: The `--logging.level.org.springframework.cloud.task=DEBUG` option lets you see output that would not otherwise be written to the console. Because Spring Cloud Task uses an in-memory database to store its results (and that in-memory database is destroyed at the end of the run), the `DEBUG` option is the only way to see the output from the timestamp task.
+
+The timestamp application shows the following output (in the midst of much other output):
+
+[source,bash]
+----
```

Comment:
Also with tasks you get a line just prior to the timestamp that looks something like this:
`2018-03-14 13:47:16.659 DEBUG 78382 --- [           main] o.s.c.t.r.support.SimpleTaskRepository   : Creating: TaskExecution{executionId=0, parentExecutionId=null, exitCode=null, taskName='application', startTime=Wed Mar 14 13:47:16 EDT 2018, endTime=null, exitMessage='null', externalExecutionId='null', errorMessage='null', arguments=[]}`
for example: 
```
2018-03-14 13:48:28.240 DEBUG 78490 --- [           main] o.s.c.t.r.support.SimpleTaskRepository   : Creating: TaskExecution{executionId=0, parentExecutionId=null, exitCode=null, taskName='application', startTime=Wed Mar 14 13:48:28 EDT 2018, endTime=null, exitMessage='null', externalExecutionId='null', errorMessage='null', arguments=[]}
2018-03-14 13:48:28.250  INFO 78490 --- [           main] i.s.simpletask.SimpletaskApplication     : Started SimpletaskApplication in 1.185 seconds (JVM running for 1.759)
Hello, World
2018-03-14 13:48:28.256 DEBUG 78490 --- [           main] o.s.c.t.r.support.SimpleTaskRepository   : Updating: TaskExecution with executionId=1 with the following {exitCode=0, endTime=Wed Mar 14 13:48:28 EDT 2018, exitMessage='null', errorMessage='null'}

```

Reply:
I added it. Thanks. :)